src/Dapr.Workflow/DaprWorkflowClient.cs

```diff
@@ -0,0 +1,310 @@
+        /// </summary>
+        /// <param name="innerClient">The Durable Task client used to communicate with the Dapr sidecar.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="innerClient"/> is <c>null</c>.</exception>
+        public DaprWorkflowClient(DurableTaskClient innerClient)
+        {
+            this.innerClient = innerClient ?? throw new ArgumentNullException(nameof(innerClient));
+        }
+
+        /// <summary>
+        /// Schedules a new workflow instance for execution.
+        /// </summary>
+        /// <param name="name">The name of the workflow to schedule.</param>
+        /// <param name="instanceId">
+        /// The unique ID of the workflow instance to schedule. If not specified, a new GUID value is used.
+        /// </param>
+        /// <param name="startTime">
+        /// The time when the workflow instance should start executing. If not specified or if a date-time in the past
+        /// is specified, the workflow instance will be scheduled immediately.
+        /// </param>
+        /// <param name="input">
+        /// The optional input to pass to the scheduled workflow instance. This must be a serializable value.
+        /// </param>
+        public Task<string> ScheduleNewWorkflowAsync(
+            string name,
+            string? instanceId = null,
+            object? input = null,
+            DateTime? startTime = null)
+        {
+            StartOrchestrationOptions options = new(instanceId, startTime);
+            return this.innerClient.ScheduleNewOrchestrationInstanceAsync(name, input, options);
+        }
+
+        /// <summary>
+        /// Fetches runtime state for the specified workflow instance.
+        /// </summary>
+        /// <param name="instanceId">The unique ID of the workflow instance to fetch.</param>
+        /// <param name="getInputsAndOutputs">
+        /// Specify <c>true</c> to fetch the workflow instance's inputs, outputs, and custom status, or <c>false</c> to
+        /// omit them. Defaults to false.
+        /// </param>
+        public async Task<WorkflowState> GetWorkflowStateAsync(string instanceId, bool getInputsAndOutputs = false)
+        {
+            OrchestrationMetadata? metadata = await this.innerClient.GetInstancesAsync(
+                instanceId,
+                getInputsAndOutputs);
+            return new WorkflowState(metadata);
+        }
+
+        /// <summary>
+        /// Waits for a workflow to start running and returns a <see cref="WorkflowState"/> object that contains metadata
+        /// about the started workflow.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// A "started" workflow instance is any instance not in the <see cref="WorkflowRuntimeStatus.Pending"/> state.
+        /// </para><para>
+        /// If a workflow instance is already running (or completed) when this method is called, the method will return immediately.

```

Comment:
I agree it's confusing. I'll reword it to something simpler. However, I don't want to list the states since that list is large and may change over time. Let me know if this rewording works for you.